app/src/main/java/com/ikarabulut/GameSettings.java

```diff
@@ -32,35 +34,93 @@ public void getAllSettings() {
 
     public void setEvolutions() {
         display.evolutionsPrompt();
-        evolutions = Integer.parseInt(input.nextLine());
+        try {
+            evolutions = Integer.parseInt(input.nextLine());
+        } catch (Exception e) {
+            System.err.print("An error occurred, please enter an integer. Try again: ");
+            setEvolutions();
+        }
     }
 
+
     public void setRows() {
         display.rowsPrompt();
-        rows = Integer.parseInt(input.nextLine());
+        try {
+            intSelection = Integer.parseInt(input.nextLine());
+        } catch (Exception ex) {
+            System.err.print("Your input is invalid. Please try again: ");
+            setRows();
+        } finally {
+            try {
+                if (intSelection > 25) {
+                    throw new IllegalArgumentException("Please limit the number of columns to under 25. Please re-enter your selection: ");
+                } else {
+                    rows = intSelection;
+                }
+            } catch (IllegalArgumentException ex) {
+                System.err.print(ex.getMessage());
+                setRows();
+            }
+        }
     }
 
     public void setColumns() {
         display.columnsPrompt();
-        columns = Integer.parseInt(input.nextLine());
+        try {
+            intSelection = Integer.parseInt(input.nextLine());
+        } catch (Exception ex) {
+            System.err.print("Your input is invalid. Please try again: ");
+            setColumns();
+        } finally {
+            try {
+                if (intSelection > 25) {
+                    throw new IllegalArgumentException("Please limit the number of columns to under 25. Please re-enter your selection: ");
+                } else {
+                    columns = intSelection;
+                }
+            } catch (IllegalArgumentException ex) {
+                System.err.print(ex.getMessage());
+                setColumns();
+            }
+        }
     }
 
     public void setAliveSymbol() {
         display.aliveSymbolPrompt();
-        aliveSymbol = input.nextLine();
+        String selection = input.nextLine();
+        try {
+            if (selection.length() > 1) {
+                throw new InvalidSymbolException("Please limit the symbol to 1 character. Please try again: ");
+            } else {
+                aliveSymbol = selection;
+            }
+        } catch (InvalidSymbolException ex) {
+            System.err.print(ex.getMessage());
+            setAliveSymbol();
+        }
     }
 
-    public void setDeadSymbol() {
+    public void setDeadSymbol() throws InvalidSymbolException {
```

Comment:
Does this need to be typed as throwing an exception? It looks like you're handling any exceptions that get thrown, so I don't think you'll need to have this here. @jramnani ?

Reply:
Since you are catching the exception and not re-throwing it within the method, that exception can be removed from the signature.  As well as the other declaration in this class.

Reply:
This was solved in my other PR. John and I talked through this